src/gui/mainwindow.cpp

```diff
@@ -1095,15 +1102,32 @@ void MainWindow::dropEvent(QDropEvent *event)
         files = event->mimeData()->text().split('\n');
     }
 
-    // Add file to download list
+    // differentiate ".torrent" files and others
+    QStringList torrentFiles, otherFiles;
+    foreach (const QString &file, files) {
+        if (file.endsWith(".torrent", Qt::CaseInsensitive))
+            torrentFiles << file;
+        else
+            otherFiles << file;
+    }
+
+    // Download torrents
     const bool useTorrentAdditionDialog = AddNewTorrentDialog::isEnabled();
-    foreach (QString file, files) {
+    foreach (const QString &file, torrentFiles) {
         qDebug("Dropped file %s on download list", qPrintable(file));
-        if (useTorrentAdditionDialog)
+        if (useTorrentAdditionDialog) {
             AddNewTorrentDialog::show(file, this);
+        }
         else
             BitTorrent::Session::instance()->addTorrent(file);
     }
+    if (!torrentFiles.isEmpty()) return;
```

Comment:
2 reasons that I can think of:
1. the user might be confused when he saw 2 different dialogs pop up at the same time, he might want to download or upload but not 2 at the same time, I doubt anyone will mix download/upload together.
2. the AddNewTorrentDialog is a modal dialog which could be a problem (dialogs overlap and user cannot switch to another) when multiple dialogs pop up at the same time.